### DISCUSSION TITLE
Add `locale` graphql field to collections/global

### DISCUSSION THREAD

**Discussion by adamalfredsson** [payloadcms/payload#1522] [Feature Requests & Ideas]
**TLDR: Add a locale graphql field so that you can determine if the requested resource is of the requested or fallback locale**

Consider this localization config:

```
{
  localization: {
    defaultLocale: "en",
    locales: ["en", "de"],
  }
}
```

1. I have a "Pages" collection with a slug field
2. I have a "Navigation" global with a links relationship field that references "Pages"
3. There exists one page with slug field value "english1" and "deutsch1" for each locale.
4. I save a new page without a "de" locale, with a slug field value of "english2". If I would query this:

```
{
  Pages(locale: de, where: { slug: { exists: true } }) {
    docs {
      slug
    }
  }
  Navigation(locale: de) {
    links {
      slug
    }
  }
}
```

I would get this:

```json
{
  "data": {
    "Pages": {
      "docs": [
        {
          "slug": "deutsch1"
        }
      ]
    },
    "Navigation": {
      "links": [
        {
          "slug": "deutsch1"
        },
        {
          "slug": "english2"
        }
      ]
    }
  }
}
```

I generate all paths statically like this:

```
=> en/english1
=> de/deutsch1
=> en/english2
```

When I render the links in the navigation, I would like the links to reference the localized page or fallback to the default "en" locale. However, given the information I get in the "Navigation" query data, I'm unable to determine if it's a localized or fallback page. If there was a built-in "locale" field on "Pages" I could query, my life would be so much better, e.g.:

```
{
  Navigation(locale: de) {
    links {
      slug
      locale
    }
  }
}
```

```json
{
  "data": {
    "Navigation": {
      "links": [
        {
          "slug": "deutsch1",
          "locale": "de"
        },
        {
          "slug": "english2",
          "locale": "en"
        }
      ]
    }
  }
}
```

**Comment by ssyberg** [comment]
I'm also finding this is crucial for dealing with Apollo cache issues, any idea if/when this will get implemented?

**Comment by l-you** [comment]
Hi @ssyberg did you managed to solve this? I am also looking for a "locale" field for partial translation of website.
